### ISSUE TITLE
How to make it so that tdlib operates like a daemon?

### ISSUE THREAD

**Issue by bangonkali** [egramtel/tdsharp#139]
Is there any sample in C# such that `tdlib` operates like a daemon and continues to update `db.sqlite` with new data as updated information is received? 

Somehow [this example from tdlib](https://github.com/tdlib/td/blob/master/example/cpp/td_example.cpp#L61-L67) can do `client_manager_->receive(10)`. 

```c++
  void loop() {
    while (true) {
      if (need_restart_) {
        restart();
      } else if (!are_authorized_) {
        process_response(client_manager_->receive(10));
      } else {
```

Is there a way to mirror this [Using strongly typed APIs](https://github.com/egramtel/tdsharp#using-strongly-typed-apis)? 

Or do I have to use the untyped version to be able to make some sort of daemon approach?

**Comment by ForNeVeR** [comment]
I don't think we support the `ClientManager`, yet. This is out of scope of "normal" MTProto API, since this is just a C++ class (not sure it has C wrappers around it to be available for P/Invoke).

**Comment by ForNeVeR** [comment]
But I can't see how it relates to your daemon question. What do you mean by that? AFAIK, tdlib just normally updates its database during execution; there's no special "daemon approach" or something like that.

**Comment by bangonkali** [comment]
Hi @ForNeVeR , 

> But I can't see how it relates to your daemon question. What do you mean by that? 

I have created a simple application in CLI however I have been left in a request and response model. I would like to know if it's possible to have a concurrent approach to request & response wherein the CLI will not shutdown but always receive updates.

Can you recommend a simple pattern for this? 

For now when my cli app for example lists chats and messages in chats, the application stops after completion. What I'm looking for is to keep tdlib awake and keep on receiving updates and my cli app can then just listen to the updates indefinitely.

I may be missing something fundamental here though. 

I will look in to `ClientManager` as it seems you may be suggesting something here but I don't understand it full yet at the moment. 😅

**Comment by ForNeVeR** [comment]
The usual pattern for daemon-like code in C# is to use a [Generic Host](https://docs.microsoft.com/en-us/dotnet/core/extensions/generic-host), that has no relation to tdlib's `ClientManager` whatsoever.
